src/Fmt_ast.ml

```diff
@@ -3710,16 +3740,47 @@ and fmt_structure c ctx itms =
           | Pmod_ident _ -> true
           | _ -> false
```

Comment:
For readability, it would be good to refactor `has_doc` to a separate function, probably in `Ast`, and generalized from structure_item to Ast.t, with the implementation for at least the signature_item added.